### PR TITLE
docs: add back to top button

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
Added a back to top button on docs pages for below 1400px screen size and gave it a classname 'scroll-up-btn', added the scroll up button related JS (made a new file) and CSS.

<details>
<summary>for light mode</summary>

Screenshot 2023-03-12 144538

</details>

<details>
<summary>for dark mode</summary>

Screenshot 2023-03-12 144603

</details>
#### Is there anything you'd like reviewers to focus on?
PR is related to the issue #16932.

### CODE REVIEW

File: docs/src/assets/scss/docs.scss
Comment (snitin315) [inline docs/src/assets/scss/docs.scss:170]: Can we use colors to match the theme?
https://github.com/eslint/eslint/blob/75df535681d15d7d685468d637945a200301f9ee/docs/src/assets/scss/tokens/themes.scss#L116

Comment (Tanujkanti4441) [reply docs/src/assets/scss/docs.scss:170]: Yes, but instead of using the css variables for colors can i use rgba form (the way it's been already used) of theme colors so i can make them transparent a little because with opacity property the arrow icon of button is also becoming faded.

**Review by kecrily — COMMENTED** [review]
I think the accessibility in light mode is not good enough, can you improve it? Like modifying the color of ↑.

**Comment by Tanujkanti4441** [comment]
> I think the accessibility in light mode is not good enough, can you improve it? Like modifying the color of

> I think the accessibility in light mode is not good enough, can you improve it? Like modifying the color of ↑.

Done.

**Comment by harish-sethuraman** [comment]
Can you add a sample page url in PR where I can see scroll to top 🤔 Im not able to see it?

**Comment by Tanujkanti4441** [comment]
> Can you add a sample page url in PR where I can see scroll to top 🤔 Im not able to see it?

I think button is on every page of docs just need to scroll down to make it appear. 
https://deploy-preview-16979--docs-eslint.netlify.app/rules/

**Comment by amareshsm** [comment]
Can't we achieve this same behavior without using js?

I think we achieve this using CSS which works well in `no-js` mode (when js is disabled):
positioning the scroll to the top button - `position: sticky`
for checking device-width - media query 
scrolling to the top - a link that targets an ID on the `<html>` element.

Thoughts?

**Comment by kecrily** [comment]
> Can't we achieve this same behavior without using js?
> 
> 
> 
> I think we achieve this using CSS which works well in `no-js` mode (when js is disabled):
> 
> positioning the scroll to the top button - `position: sticky`
> 
> for checking device-width - media query 
> 
> scrolling to the top - a link that targets an ID on the `<html>` element.
> 
> 
> 
> Thoughts?

Yes, I think `no-js` support is so cool. And it is true that we can achieve this feature without js.

**Comment by Tanujkanti4441** [comment]
> Can't we achieve this same behavior without using js?
> 
> I think we achieve this using CSS which works well in `no-js` mode (when js is disabled): positioning the scroll to the top button - `position: sticky` for checking device-width - media query scrolling to the top - a link that targets an ID on the `<html>` element.
> 
> Thoughts?

Agree with this, but i am not sure about how `Position: sticky` will help the button to achieve the same behavior (appear after scrolling) because for that, button should be placed on some where below from the bottom of the view-port and as i tried `position: sticky` and not able to move it any where from its default position (bottom of the page and aligned to the left side).
is there something we can do or am i missing something.

**Comment by harish-sethuraman** [comment]
May be we can remove the sticky one when JS is enabled after scrolling a bit. If not lets leave it as such. Also for scrolling to top lets use `id` based approach without any JS.

**Comment by Tanujkanti4441** [comment]
> May be we can remove the sticky one when JS is enabled after scrolling a bit. If not lets leave it as such. Also for scrolling to top lets use `id` based approach without any JS.

Done! The back to button is now  functioning without js.

File: docs/src/_includes/layouts/base.html
Comment (harish-sethuraman) [inline docs/src/_includes/layouts/base.html:2]: lets rename `scroll_to_top` with something else? like page-start or some other that makes sense

File: docs/src/_includes/layouts/doc.html
Comment (harish-sethuraman) [inline docs/src/_includes/layouts/doc.html:105]: We are using id for js and class for css can we use id for both please?

Comment (harish-sethuraman) [reply docs/src/_includes/layouts/doc.html:105]: and Im not sure if we are supposed to use up arrow icon like this. Can we rather replace it with svg? so that it doesn't break?

File: docs/src/assets/js/scroll-up-btn.js
Comment (harish-sethuraman) [inline docs/src/assets/js/scroll-up-btn.js:6]: isn't document available why are we doing `this.document`?

File: docs/src/assets/scss/docs.scss
Comment (harish-sethuraman) [inline docs/src/assets/scss/docs.scss:174]: lets use css variables for colors `--color-primary-800` is the variable you are looking for

File: docs/src/assets/scss/docs.scss
Comment (harish-sethuraman) [inline docs/src/assets/scss/docs.scss:186]: lets use css variables here also

File: docs/src/assets/scss/docs.scss
Comment (Tanujkanti4441) [reply docs/src/assets/scss/docs.scss:174]: the rgba form of these theme color is used to make them transparent a bit. didn't use opacity because it is also making the arrow transparent as well.

Comment (Tanujkanti4441) [reply docs/src/assets/scss/docs.scss:174]: I thought that It would be good to keep this button transparent so it doesn't feel like that it is blocking something in the page.

Comment (harish-sethuraman) [reply docs/src/assets/scss/docs.scss:174]: Most of the time the user should be able to scroll a little more to see content below the button? I think its fine to not include opacity. adding opacity will just make it harder to see content below.

File: docs/src/assets/scss/docs.scss
Comment (harish-sethuraman) [inline docs/src/assets/scss/docs.scss:170]:
```suggestion
    color: var(--body-background-color);
```

File: docs/src/assets/scss/docs.scss
Comment (harish-sethuraman) [inline docs/src/assets/scss/docs.scss:175]:
```suggestion
```

**Review by harish-sethuraman — CHANGES_REQUESTED** [review]
Lets make sure we add `display: none` for this element in the `print.scss` so that we dont see the arrow in print preview :). Other than that everything looks perfect thanks!

**Review by harish-sethuraman — APPROVED** [review]
LGTM. 
cc: @eslint/website-team friendly ping for review

File: docs/src/assets/scss/docs.scss
Comment (kecrily) [reply docs/src/assets/scss/docs.scss]: I think not include opacity is better, the user can see a little through the button but if it is not completely clear it will only cause misleading.

Comment (Tanujkanti4441) [reply docs/src/assets/scss/docs.scss]: Yes, opacity is removed now completely.

**Review by kecrily — APPROVED** [review]
LGTM

**Comment by nzakas** [comment]
@kecrily just a reminder: if you approve a PR and don't merge it, please explain why. Right now I'm not sure if you're expecting something else to happen or not.

File: docs/src/assets/js/scroll-up-btn.js
Comment (harish-sethuraman) [inline docs/src/assets/js/scroll-up-btn.js:4]: I have a suggestion here. This width check can be removed and media query can be used instead to show or hide scroll button for desktop. In `no-js` right now it will show in desktop device too?

**Comment by kecrily** [comment]
> @kecrily just a reminder: if you approve a PR and don't merge it, please explain why. Right now I'm not sure if you're expecting something else to happen or not.

@nzakas I thought that my request for a review of snitin315 had shown the reason why I didn't merge it. But it doesn't matter. I will indicate this separately next time.

File: docs/src/_includes/layouts/base.html
Comment (amareshsm) [inline docs/src/_includes/layouts/base.html:157]:
```suggestion

```

File: docs/src/_includes/layouts/doc.html
Comment (amareshsm) [inline docs/src/_includes/layouts/doc.html:107]:
```suggestion
 <div class="scroll-up-container">
    <a id="scroll-up-btn" href="#site_top">
        <svg fill="none" height="24" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" viewBox="0 0 24 24" width="24"><line x1="12" x2="12" y1="19" y2="5"/><polyline points="5 12 12 5 19 12"/></svg>
    </a>
</div>
```

File: docs/src/assets/js/scroll-up-btn.js
Comment (amareshsm) [inline docs/src/assets/js/scroll-up-btn.js:13]:
```suggestion

```

File: docs/src/assets/scss/print.scss
Comment (amareshsm) [inline docs/src/assets/scss/print.scss:211]:
```suggestion
.scroll-up-container {
```

File: docs/src/assets/scss/docs.scss
Comment (amareshsm) [inline docs/src/assets/scss/docs.scss:183]:
```suggestion
.scroll-up-container {
    display: none;
    position: absolute;
    top: 0;
    right: 30px;
    height: 100%;

    &::before {
        content: "";
        display: block;
        height: 120vh;
        pointer-events: none;
    }
    #scroll-up-btn {
        position: sticky;
        top: 80vh;
        cursor: pointer;
        width: 50px;
        height: 50px;
        display: flex;
        font-size: 1.5rem;
        border-radius: 50%;
        color: var(--body-background-color);
        text-decoration: none;
        justify-content: center;
        align-items: center;
        background-color: var(--link-color);
        transition: all 1s ease-in;
    }
}

@media screen and (max-width: 1400px) {
    .scroll-up-container {
        display: block;
    }
}

```

**Comment by amareshsm** [comment]
I tried to achieve the same behavior without using js. Pls have a look. I checked locally It works.

**Comment by Tanujkanti4441** [comment]
> I tried to achieve the same behavior without using js. Pls have a look. I checked locally It works.

@amareshsm i tried it as the same way as yours and it is working fine on desktop but not on mobile/tablet because in browser apps, on scrolling, the viewable screen height is also changing because browsers widget (like for changing the tabs and URL input) is disappearing on scrolling up and appearing on scrolling down,
and this behavior is affecting the position of back to top button.
is it same for you as well ?
if yes can we do something for this.

<details>
<summary>Screenshot</summary>

Screenshot_2023-03-25-16-57-46-01
</details>

<details>
<summary>Screenshot</summary>

Screenshot_2023-03-25-16-57-17-90
</details>

**Comment by amareshsm** [comment]
> > I tried to achieve the same behavior without using js. Pls have a look. I checked locally It works.
> 
> @amareshsm i tried it as the same way as yours and it is working fine on desktop but not on mobile/tablet because in browser apps, on scrolling, the viewable screen height is also changing because browsers widget (like for changing the tabs and URL input) is disappearing on scrolling up and appearing on scrolling down, and this behavior is affecting the position of back to top button. is it same for you as well ? if yes can we do something for this.
> 
> Screenshot
> Screenshot

Yes it is happening based on the device viewport height it is changing. We can write media queries based on height and handle it. For now we can go ahead with current approach.

**Review by amareshsm — APPROVED** [review]
LGTM. Waiting for code owner review.

**Review by snitin315 — APPROVED** [review]
LGTM. Thanks for contributing.

**Pull request merged**
